kazoo/recipe/lock.py

```diff
@@ -43,18 +43,31 @@ class Lock(object):
     Note: This lock is not *re-entrant*. Repeated calls after already
     acquired will raise a ``RuntimeError``.
 
+    This is an exclusive lock. For a read/write lock, see :method:`WLock` and
+    :method:`RLock`.
+
     """
-    _NODE_NAME = '__lock__'
 
-    def __init__(self, client, path, identifier=None):
+    def __init__(self, client, path, identifier=None, node_name="__lock__",
```

Comment:
I'd almost rather have ``class Lock(object):`` -> ``class _Lock(object):``

And then only have these keyword arguments on ``_Lock``; what do u think? Even though u are saying 'don't pass them' it seems like we could do better...